python/TestHarness/TestHarness.py

```diff
@@ -235,85 +240,148 @@ def launchTest(self, find_only, file, dirpath, testers):
-                        self.handleTestStatus(tester)
-                    elif status == tester.bucket_deleted and self.options.extra_info:
-                        self.handleTestStatus(tester)
-                self.runner.jobSkipped(tester.parameters()['test_name'])
-
-        # Close the tests.cluster file
-        if self.options.cluster_handle is not None:
-            self.options.cluster_handle.close()
-            self.options.cluster_handle = None
-
-        # Return the final tester/command (sufficient to run all tests)
-        return (tester, command)
+            # Double the alloted time for tests when running with the valgrind option
+            tester.setValgrindMode(self.options.valgrind_mode)
+
+            # When running in valgrind mode, we end up with a ton of output for each failed
+            # test.  Therefore, we limit the number of fails...
+            if self.options.valgrind_mode and self.num_failed > self.options.valgrind_max_fails:
+                tester.setStatus('Max Fails Exceeded', tester.bucket_fail)
+            elif self.num_failed > self.options.max_fails:
+                tester.setStatus('Max Fails Exceeded', tester.bucket_fail)
+            elif tester.parameters().isValid('error_code'):
+                tester.setStatus('Parser Error', tester.bucket_skip)
+            else:
+                should_run = tester.checkRunnableBase(self.options, self.checks, self.test_list)
+                # check for deprecated tuple
+                if type(should_run) == type(()):
+                    (should_run, reason) = should_run
+                    if not should_run:
+                        reason = 'deprected checkRunnableBase #8037'
+                        tester.setStatus(reason, tester.bucket_skip)
+
+                # Regarding (or self.options.processingPBS:) below:
+                # When processing PBS results, there are no "skipped" tests present in the PBS batch file.
+                # Therefore should_run should always be true. This unfortunately has the undesired result
+                # of not listing any skipped tests during the 'Final Test Result' stage.
+
+                # We are doing this because the 'launchTest' method loops through every tester generated.
+                # while the PBS batch file is only storing information on tests that have run. Without
+                # this addition, individual skipped tests that are apart of _this_ test in the list of
+                # testers as a whole get repeatedly printed as we iterate over each PBS test.
+
+                # TODO: Try and figure out a way to print these skipped tests only once when processing
+                # PBS batch history.
+                if should_run or self.options.processingPBS:
+                    command = tester.getCommand(self.options)
+
+                    # Handle PBS processing
+                    # TODO: figure out how to allow getCommand do this
```

Comment:
getCommand can not do this. Testers such as AnalyzeJacobian have their own getCommand method which overrides the base class. And we need to override them so...